Fix/error code on fail

gkapfham opened the pull request:
<!-- TODO: Replace the title below -->
# Ensure that a Non-Zero Error Code is Created for Failing Checks

## Description

Right now the program does not return a non-zero exit code when some of the checks pass.

This is a problem because then a build in GitHub Actions will pass even if there are failing checks
when you run the `gatorgrade` command.

This fix ensures that a non-zero exit code, specifically `1`, will be returned.

This will cause a `gatorgrade` command in GitHub Actions to fail the build.

## Linked Issues

<!-- Fill in which issue number this PR closes, if there is none, just remove this section! -->
closes: #100 

## Type of Change

<!-- TODO: Fill in the brackets with an `x` next to all types that apply to the proposed changes -->

- [ ] Feature
- [X] Bug fix
- [ ] Documentation

## Contributors

<!-- TODO: Add your GitHub username below and the GitHub usernames of all other contributors to the proposed changes -->
- @gkapfham 

## Reminder

 - [X]  All GitHub Actions should be in a passing state before any pull request is merged.
 - [X]  All PRs must be reviewed by at least one team member and one member of the Integration team!
 - [X]  Any issues this PR closes are tagged in the description!

dluman reviewed (approved):
I've run these changes locally and produced both successful and incomplete assignments, checking system-reported exit codes produced by `gatorgrade`. In all scenarios, the correct exit code appeared.

I've reproduced correct outcome in GitHub Actions (see test builds below):

* [Failing build[(https://github.com/allegheny-college-sandbox/assign-test/runs/8047331528?check_suite_focus=true)
* [Passing build](https://github.com/allegheny-college-sandbox/assign-test/runs/8047337042?check_suite_focus=true)

gkapfham commented:
Hello @dluman, thanks for your review of this PR.

I have made several improvements since you submitted your review:

- Ensured that all test cases pass
- Add additional tools, like `isort`, for formatting and checking source code
- Improved the `pyproject.toml` file and the tasks available to developers
- Added comments to all of the new source code
- Added comments to some of the existing source code

I have also tested the project on this branch by using `pipx` to install with the following command:

`pipx install git+https://github.com/GatorEducator/gatorgrade.git@fix/error-code-on-fail`

This allows me to install `gatorgrade` directly from this branch and then test it in projects.

When I use this version of `gatorgrade` I get the exit codes that I would expect.

As such, I'm going to merge this PR to see what happens in regards to a deploy that releases the package to PyPI. If the automated deploy to PyPI does not work I will attempt to do a manual deploy and then check everything to ensure that we see the expected output in starter and solution repositories and on local and GitHub Actions.
